crates/telemetry/src/config.rs

```diff
@@ -0,0 +1,78 @@
+use envconfig::Envconfig;
+use opentelemetry::KeyValue;
+use opentelemetry_semantic_conventions::resource::SERVICE_VERSION;
+
+/// Provides configuration for the telemetry system.
+///
+/// Largely consists of OpenTelemetry variables defined
+/// [here](https://opentelemetry.io/docs/specs/otel/configuration/sdk-environment-variables/#general-sdk-configuration)
+/// but also has some Spin specific configuration.
+#[derive(Envconfig, Debug)]
+pub struct Config {
+    /// Disables OpenTelemetry for all signals (traces and metrics).
+    #[envconfig(from = "OTEL_SDK_DISABLED", default = "false")]
+    pub otel_sdk_disabled: bool,
+
+    /// Value of the `service.name` resource attribute.
+    #[envconfig(from = "OTEL_SERVICE_NAME", default = "spin")]
```

Comment:
It occurs to me that we probably want the `service.name` to be whatever the user set as the app name in their manifest rather than just `Spin`. This would require us to read the manifest immediatley at the start of `spin.rs` since reloading the tracing subscriber isn't working. Are we okay with that?

Side note: How are we going to handle hot-swapping the OTEL resource in cloud?

Reply:
Lets hold off on setting the service.name to the app name automatically until we understand the impact on shim users.

Reply:
Also I _think_ this can handle at least this var for you: https://docs.rs/opentelemetry_sdk/0.22.0/opentelemetry_sdk/resource/struct.EnvResourceDetector.html (with https://docs.rs/opentelemetry_sdk/0.22.0/opentelemetry_sdk/struct.Resource.html#method.from_detectors)